opencog/cython/opencog/atomspace.pxd

```diff
@@ -88,6 +88,7 @@ cdef extern from "opencog/atomspace/Handle.h" namespace "opencog":
     cdef cppclass cHandle "opencog::Handle":
         cHandle()
         cHandle(UUID)
+        cHandle(void*)
```

Comment:
I definitely understand what you were saying. I too thought there must be some way to do that but there isn't because you can't pass AtomPtr (shared_ptr) objects as Python objects into Cython's Python object initializers, which is where these initializers are called from. 

The problem is that the Python Atom object required for the tuple for the call of the Python function has to be initialized using Python PyObject* object arguments. The Python PyObject* objects have very limited types. Cython lets you do a lot but it doesn't change the semantics of Python objects and it doesn't let you pass anything by reference.

The only device for handling more complicated structures that need to be passed through the PyObject* argument interface is a pointer. So you can pass pointers to the shared_ptrs reliably by converting them to PyLong objects first

In fact, the only reliable way to pass pointers across the interface from C++ to the Python object initializers is by passing them as void*s converted to Python variables using the two routines provided for this purpose:

PyLong_AsVoidPtr
PyLong_FromVoidPtr